cpp/memilio/io/auto_serialize.h

```diff
@@ -0,0 +1,218 @@
+/*
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Rene Schmieding
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef MIO_IO_AUTO_SERIALIZE_H_
+#define MIO_IO_AUTO_SERIALIZE_H_
+
+#include "memilio/io/io.h"
+#include "memilio/utils/metaprogramming.h"
+
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace mio
+{
+
+/**
+ * @brief Pair of name and value used for auto-(de)serialization.
+ *
+ * This object holds a view of a name and reference of a value. Mind their lifetime!
+ * @tparam ValueType The (non-cv, non-reference) type of the value. 
+ */
+template <class ValueType>
+struct NVP {
+    using Type = ValueType&;
+
+    const std::string_view name;
+    Type value;
+
+    /**
+     * @brief Create a (name, value) pair.
+     *
+     * @param n A view of the name. 
+     * @param v A non-const lvalue reference to the value. 
+     */
+    explicit NVP(const std::string_view n, Type v)
+        : name(n)
+        , value(v)
+    {
+    }
+
+    NVP()                      = delete;
+    NVP(const NVP&)            = default;
+    NVP(NVP&&)                 = default;
+    NVP& operator=(const NVP&) = delete;
+    NVP& operator=(NVP&&)      = delete;
+};
+
+/**
+ * @brief Provide names and values for auto-(de)serialization.
+ *
+ * This function packages the class name and a name-value pair for each class member together to define both a
+ * serialize and deserialize function (with limited features).
+ *
+ * Note that auto-serialization requires that all class members participate in serialization, and that
+ * each class member is (auto-)serializable and assignable.
+ * 
+ * @tparam Targets List of each class member's type.
+ * @param class_name The name of the class to auto-serialize.
+ * @param class_members A name-value pair (NVP) for each class member.
+ * @return Collection of all name views and value references used for auto-(de)serialization. 
+ */
+template <class... Targets>
+[[nodiscard]] inline auto make_auto_serialization(const std::string_view&& class_name, NVP<Targets>&&... class_members)
```

Comment:
I think I prefer `default_serialization` (similar to default special member functions, just a bit more work)

Reply:
Do you mean as a name over auto_serialization? I think that works fine, I am not particularly attached to the name.
Should I keep the dash in text, e.g. "default-(de)serialization"?

(Note that `make_auto_serialization` is (or will be) replaced by Members().add().)

Reply:
yes, default_serialization instead of auto_serialization (and DefaultSerialization instead of AutoSerialization, and default serialization in the documentation, without dash I think).